### PR TITLE
Issue 5197: Limit number of transactions in a commit batch

### LINKED ISSUE #5197: Limit number of transactions in a batch while processing commits
**Problem description**
Add an upper limit on number of transactions that will be batched together and committed. 

**Problem location**
Commit Transaction workflow

**Suggestions for an improvement**
Add an upper limit on number of transactions to be processed in a batch.

### PR DESCRIPTION
**Pull request by shiveshr**
Signed-off-by: Shivesh Ranjan <shivesh.ranjan@gmail.com>

**Change log description**  
Limit number of transactions in a commit batch.

**Purpose of the change**  
Fixes #5197 

**What the code does**  
We add a new "limit" paramter to store.startCommitTransaction method which only starts a batch equal to the size of commitBatch. 

**How to verify it**  
Unit test added

### CODE REVIEW

File: controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java [hunk 1]
```diff
@@ -1673,7 +1677,11 @@ boolean compareWriterPositions(Map<Long, Long> position1, Map<Long, Long> positi
             txnIds.add(x.getValue());
         });
 
-        return getTransactionRecords(epoch, txnIds).thenAccept(txns -> {
+        AtomicInteger from = new AtomicInteger(0);
+        Function<Integer, Integer> tillFunc = f -> f + limit < txnIds.size() ? f + limit : txnIds.size();
```
Comment (shrids) [inline controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java:1681]: Math.min () can be used here ?

File: controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java [hunk 2]
```diff
@@ -1683,6 +1691,9 @@ boolean compareWriterPositions(Map<Long, Long> position1, Map<Long, Long> positi
                         if (txnRecord.getCommitOrder() == order) {
                             // if entry matches record's position then include it
                             transactionsMap.put(txnId, txnRecord);
+                            if (transactionsMap.size() >= limit) {
```
Comment (shrids) [inline controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java:1694]: do we require this check given that the txnIds are not greater than the passed limit ?

**Comment by shrids** [comment]
System test run for master @ `job/test-pravega-continuous/3186/`

File: controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java (hunk 2 above)
Comment (shiveshr) [reply controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java:1694]: transactionsMap is actually accumulating filtered txns from the txnIds across all iterations and the limit s really to be imposed on the number of entries we put in this map as that will be sent back to the caller to use as the batch of transactions

**Review by shrids — APPROVED** [review]
+1 LGTM on code inspection. 
This code should be merged post the system test run.

**Pull request merged**